src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php

```diff
@@ -295,13 +295,14 @@ private static function analyzeNonDivOperands(
             && $right_type_part instanceof TLiteralInt
             && ($left instanceof PhpParser\Node\Scalar
                 || $left instanceof PhpParser\Node\Expr\ConstFetch
-                || $left instanceof PhpParser\Node\Expr\ClassConstFetch)
+                || $left instanceof PhpParser\Node\Expr\ClassConstFetch
+                || $left instanceof PhpParser\Node\Expr\BinaryOp)
```

Comment:
Shouldn't this be any expression really? E.g. `(true ? 1 : 2) | 22` would not have `$left` as binary op, but it still can be calculated, I suppose.

Reply:
Well, we saw a good example of 1 | 2 | 4 in the form of flags and we need Psalm to understand it to be coherent. I'm not sure if it's worth adding complexity and risking edge cases for  `(true ? 1 : 2) | 22`. I'd be open to consider this if we find a real case where it change the outcome of psalm's analysis

Reply:
I can imagine things like ` & ~$this->flagMask `, but you're right, we can fix it when we see it.